server/api/controllers/document-controller.ts

```diff
@@ -0,0 +1,269 @@
+      If doesn't then it creates it.
+      Then it uploads the file to the path and rewrites the json file.
+    */
+  static async processDocumentUpload(request: DocumentUploadRequest) {
+    const { documents, errors, res, path, documentJSONPathEndPoint } = request;
+
+    try {
+      for (const document of documents) {
+        if (!existsSync(path)) {
+          mkdirSync(path, { recursive: true });
+        }
+        await document.file?.mv(`${path}/${document.fileName}`);
+        this.logger.info(`${document.fileName} uploaded to ${path}`);
+      }
+
+      this.rewriteDocumentJson(documentJSONPathEndPoint);
+    } catch (err) {
+      this.logger.error(err);
+    }
+    const urlSearchParams = new URLSearchParams();
+    errors.forEach((item) => urlSearchParams.append("errors", item));
+    res.send({ status: "success", errors: errors });
+  }
+
+  /*
+      Delete the image from the given path and updates the json file.
+    */
+  static async processDocumentDelete(request: DocumentDeleteRequest) {
+    const { fileName, path, res, documentJSONPathEndPoint } = request;
+    try {
+      unlinkSync(path);
+      this.logger.info(`${fileName} deleted from ${path}`);
+      this.rewriteDocumentJson(documentJSONPathEndPoint);
+      res.status(200).json({ status: "success" });
+    } catch (err) {
+      this.logger.error(err);
+      res.status(400).json({ status: "fail" });
+    }
+  }
+
+  private static generateJSON(jsonEndPath): Document[] {
+    const splitPath = jsonEndPath.split("/");
+    const url = "_hidden/document-list/" + jsonEndPath;
+    const fullPath = `server/data/${url}.json`;
+
+    if (!existsSync(fullPath)) {
+      mkdirSync(`server/data/_hidden/document-list/${splitPath[0]}`, {
+        recursive: true,
+      });
+      writeFileSync(fullPath, "[]");
+    }
+
+    const uploadedFiles: Dirent[] = readdirSync(
+      this.getDocumentPath(jsonEndPath),
+      {
+        withFileTypes: true,
+      }
+    );
+
+    let jsonStructure: Document[] = [];
+
+    ReadWriteController.getJSONDataPath(
+      url,
+      (responseCode: number, responseBody: any) => {
+        jsonStructure = responseBody;
+      }
+    );
+
+    const documents: Document[] = [];
+
+    for (const document of uploadedFiles) {
+      if (document.isFile()) {
+        documents.push({
+          fileName: document.name,
+          fileType: document.name.split(".")[1],
+          path: this.getDocumentPath(`${jsonEndPath}/${document.name}`),
+          date: this.findDate(jsonStructure, document.name),
+        });
+      }
+    }
+    return documents;
+  }
+
+  private static findDate(jsonStructure, fileName): string {
+    let fileDate;
+
+    for (const doc of jsonStructure) {
+      if (doc.fileName == fileName) {
+        fileDate = doc.date;
+        return fileDate;
+      }
+    }
+    return new Date().toJSON().slice(0, 10).replace(/-/g, "/");
```

Comment:
Is this supposed to be the fallback if the file date doesn't exist?

Reply:
yep! realistically it shouldn't ever be accessed but it's just a safe guard